### PR TITLE
fix(esp_tinyusb): Provide default descriptors for Vendor specific class

### PR DESCRIPTION
**Pull request by tore-espressif**
## Related
* Closes https://github.com/hathach/tinyusb/issues/2806
* Closes https://github.com/espressif/esp-idf/issues/14589

## TODO
- [x] Add test_app

### CODE REVIEW

File: device/esp_tinyusb/usb_descriptors.c [hunk 1]
```diff
@@ -96,20 +96,20 @@ const char *descriptor_str_default[] = {
...
-    "",
 #endif
 
 #if CONFIG_TINYUSB_NET_MODE_ECM_RNDIS || CONFIG_TINYUSB_NET_MODE_NCM
     "USB net",                               // 6. NET Interface
     "",                                      // 7. MAC
 #endif
+
+#if CFG_TUD_VENDOR
+    "Vendor specific IF",                    // 8. Vendor specific
```
Comment (roma-jam) [inline device/esp_tinyusb/usb_descriptors.c:111]:
```suggestion
    "Vendor specific",                    // 8. Vendor specific
```

File: device/esp_tinyusb/descriptors_control.c [hunk 2]
```diff
@@ -178,8 +178,8 @@ esp_err_t tinyusb_set_descriptors(const tinyusb_config_t *config)
 
     // Select FullSpeed configuration descriptor
     if (config->configuration_descriptor == NULL) {
-        // Default configuration descriptor is provided only for CDC, MSC and NCM classes
-#if (CFG_TUD_HID > 0 || CFG_TUD_MIDI > 0 || CFG_TUD_CUSTOM_CLASS > 0 || CFG_TUD_ECM_RNDIS > 0 || CFG_TUD_DFU > 0 || CFG_TUD_DFU_RUNTIME > 0 || CFG_TUD_BTH > 0)
+        // Default configuration descriptor is provided only for some classes
```
Comment (roma-jam) [inline device/esp_tinyusb/descriptors_control.c:181]:
```suggestion
        // Default configuration descriptor must be provided for the next classes
```

File: device/esp_tinyusb/descriptors_control.c [hunk 3]
```diff
@@ -192,8 +192,8 @@ esp_err_t tinyusb_set_descriptors(const tinyusb_config_t *config)
 #if (TUD_OPT_HIGH_SPEED)
     // High Speed
     if (config->hs_configuration_descriptor == NULL) {
-        // Default configuration descriptor is provided only for CDC, MSC and NCM classes
-#if (CFG_TUD_HID > 0 || CFG_TUD_MIDI > 0 || CFG_TUD_CUSTOM_CLASS > 0 || CFG_TUD_ECM_RNDIS > 0 || CFG_TUD_DFU > 0 || CFG_TUD_DFU_RUNTIME > 0 || CFG_TUD_BTH > 0)
+        // Default configuration descriptor is provided only for some classes
```
Comment (roma-jam) [inline device/esp_tinyusb/descriptors_control.c:195]: Likewise: 
```suggestion
        // Default configuration descriptor must be provided for the next classes
```

**Review by roma-jam — APPROVED** [review]
Couple of suggestions, otherwise LGTM.

**Comment by tore-espressif** [comment]
@roma-jam Thank you for the review! I updated the code and added test for the vendor class

File: device/esp_tinyusb/descriptors_control.c [hunk 4]
```diff
@@ -178,8 +178,8 @@ esp_err_t tinyusb_set_descriptors(const tinyusb_config_t *config)
 
     // Select FullSpeed configuration descriptor
     if (config->configuration_descriptor == NULL) {
-        // Default configuration descriptor is provided only for CDC, MSC and NCM classes
-#if (CFG_TUD_HID > 0 || CFG_TUD_MIDI > 0 || CFG_TUD_CUSTOM_CLASS > 0 || CFG_TUD_ECM_RNDIS > 0 || CFG_TUD_DFU > 0 || CFG_TUD_DFU_RUNTIME > 0 || CFG_TUD_BTH > 0)
+        // Default configuration descriptor must be provided for the following classes
```
Comment (leeebo) [inline device/esp_tinyusb/descriptors_control.c:181]: Does it mean if using the following classes in esp_tinyusb, the user has no chance to self-define the configuration descriptor?

**Review by leeebo — APPROVED** [review]
@tore-espressif LGTM!

File: device/esp_tinyusb/descriptors_control.c (hunk 4 above)
Comment (tore-espressif) [reply device/esp_tinyusb/descriptors_control.c:181]: It means the opposite. If the user implements HID. MIDI, DFU, BTH... classes he must provide a self-defined configuration descriptor.

For CDC, Vendor Specific, MSC and NCM classes we provide a default descriptor that the user can override with self-defined descriptor

**Pull request merged**